From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wayne Chang <waynec@nvidia.com>
Date: Tue, 3 Sep 2024 08:38:53 +0000
Subject: [PATCH 151/241] NVIDIA: SAUCE: usb: gadget: tegra-xudc: Increase
 credit HP timeout margin

Increased the CREDIT_HP_TIMER timeout value to 0x33 from the previous 0x32.
This adjustment enlarges the timeout margin and ensures that the PUT does
not enter recovery prematurely.

Bug 4779599

Change-Id: I2557bbada8ce555479c08bab9c25fb5b3ff2821c
Signed-off-by: Wayne Chang <waynec@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3250527
Reviewed-by: HaoTien Hsu <haotienh@nvidia.com>
Reviewed-by: Henry Lin <henryl@nvidia.com>
Reviewed-by: WK Tsai <wtsai@nvidia.com>
Reviewed-by: svcacv <svcacv@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
(cherry picked from commit 03f24b76101bbd1bca5ad6c89d0fed84681f09d6)

diff --git a/drivers/usb/gadget/udc/tegra-xudc.c b/drivers/usb/gadget/udc/tegra-xudc.c
index b5cf8a375d73..e425d2c0001a 100644
--- a/drivers/usb/gadget/udc/tegra-xudc.c
+++ b/drivers/usb/gadget/udc/tegra-xudc.c
@@ -3548,6 +3548,15 @@ static void tegra264_xudc_device_params_init(struct tegra_xudc *xudc)
 	val = xudc_readl(xudc, BLCG) & ~BLCG_COREPLL_PWRDN;
 	xudc_writel(xudc, val, BLCG);
 
+#define SSPX_CORE_CNT13 0x4044
+#define  SSPX_CORE_CNT13_CRDTHP_TIMER_MASK GENMASK(11, 0)
+#define  SSPX_CORE_CNT13_CRDTHP_TIMER(x) ((x) & \
+					SSPX_CORE_CNT13_CRDTHP_TIMER_MASK)
+	val = xudc_readl(xudc, SSPX_CORE_CNT13);
+	val &= ~(SSPX_CORE_CNT13_CRDTHP_TIMER_MASK);
+	val |= SSPX_CORE_CNT13_CRDTHP_TIMER(0x33);
+	xudc_writel(xudc, val, SSPX_CORE_CNT13);
+
 #define SSPX_CORE_CNT27 0x407c
 #define  SSPX_CORE_CNT27_PING_LFPS_TRPT_MASK GENMASK(29, 0)
 #define  SSPX_CORE_CNT27_PING_LFPS_TRPT(x) ((x) & \
-- 
2.43.0

